shiny/ui/_chat.py

```diff
@@ -669,32 +673,34 @@ async def _handle_error():
 
         return _stream_task
 
-    def get_latest_stream_result(self) -> str | None:
+    def get_message_stream(self) -> reactive.ExtendedTask[[], str]:
         """
-        Reactively read the latest message stream result.
+        React to changes in the latest message stream.
+
+        Reactively reads for the latest :class:`~shiny.reactive.ExtendedTask` behind the
+        latest message stream.
 
-        This method reads a reactive value containing the result of the latest
-        `.append_message_stream()`. Therefore, this method must be called in a reactive
-        context (e.g., a render function, a :func:`~shiny.reactive.calc`, or a
-        :func:`~shiny.reactive.effect`).
+        From the return value (i.e., the extended task), you can then:
+
+        1. Reactively read for the final `.result()`.
+        2. `.cancel()` the stream.
+        3. Check the `.status()` of the stream.
 
         Returns
         -------
         :
-            The result of the latest stream (a string).
+            An extended task that represents the streaming task. The `.result()` method
+            of the task can be called in a reactive context to get the final state of the
+            stream.
 
-        Raises
-        ------
-        :
-            A silent exception if no stream has completed yet.
+        Note
+        ----
+        If no stream has yet been started when this method is called, then a "mock" task
+        is returned. This mock task behaves much like a stream that hasn't yet completed,
+        except it has a `.status()` of "initial" instead of "running", and `.cancel()`
+        is a no-op.
```

Comment:
I think most of this block would be better as a comment in the code. I think the mock task is a good way to paper over the incongruity in the initial state; I also think the beauty of this method is that you can interact with it as an ExtendedTask without having to know that we're swapping out the extended task with each stream.

Here I think you could just stick with "it returns an ET with `.status()` of `"initial"`" and that it doesn't change state until a message is streamed.